Sources/TuistKit/Services/Cloud/CloudProjectDeleteService.swift

```diff
@@ -0,0 +1,74 @@
+import Foundation
+import TSCBasic
+import TuistCloud
+import TuistLoader
+import TuistSupport
+
+enum CloudProjectDeleteServiceError: FatalError, Equatable {
+    case invalidProjectName(String)
+    case projectNotFound(String)
+
+    var description: String {
+        switch self {
+        case let .invalidProjectName(projectName):
+            return "The project \(projectName) is invalid. The project should be in the format of account/project."
+        case let .projectNotFound(projectName):
+            return "The project \(projectName) was not found. Make sure it exists."
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .invalidProjectName, .projectNotFound:
+            return .abort
+        }
+    }
+}
+
+protocol CloudProjectDeleteServicing {
+    func run(
+        project: String,
+        serverURL: String?
+    ) async throws
+}
+
+final class CloudProjectDeleteService: CloudProjectDeleteServicing {
+    private let deleteProjectService: DeleteProjectServicing
+    private let listProjectsService: ListProjectsServicing
+    private let cloudURLService: CloudURLServicing
+
+    init(
+        deleteProjectService: DeleteProjectServicing = DeleteProjectService(),
+        listProjectsService: ListProjectsServicing = ListProjectsService(),
+        cloudURLService: CloudURLServicing = CloudURLService()
+    ) {
+        self.deleteProjectService = deleteProjectService
+        self.listProjectsService = listProjectsService
+        self.cloudURLService = cloudURLService
+    }
+
+    func run(
+        project: String,
+        serverURL: String?
+    ) async throws {
+        let cloudURL = try cloudURLService.url(serverURL: serverURL)
+        let projectComponents = project.components(separatedBy: "/")
+        guard projectComponents.count == 2 else { throw CloudProjectDeleteServiceError.invalidProjectName(project) }
+
+        guard let project = try await listProjectsService.listProjects(
+            serverURL: cloudURL,
+            accountName: projectComponents[0],
+            projectName: projectComponents[1]
+        ).first
+        else {
+            throw CloudProjectDeleteServiceError.projectNotFound(project)
+        }
+
+        try await deleteProjectService.deleteProject(
+            projectId: project.id,
+            serverURL: cloudURL
+        )
+
+        logger.info("Successfully deleted the \(project.slug) project.")
```

Comment:
We are using `slug` here because that's defined in the backend but I don't want slug to be a name users need to interact with. But I also don't want to change slug everywhere on the backend now 'cause that will cause a bunch of changes (although that would be preferred). Do you have an idea for a better name for the full `foo/bar` project name?